docs/modules/ROOT/pages/secretclass.adoc

```diff
@@ -0,0 +1,117 @@
+= `SecretClass`
+
+A `SecretClass` is a cluster-global Kubernetes resource that defines a category of secrets that the Secret Operator knows how to provision.
+
+This is intended to provide an abstraction between how the secret is used ("I need a certificate for my cluster's TLS PKI") and how
+it is provisioned (automatically and generated by the operator's internal CA, provisioned by the cluster administrator, or provisioned by an
+external service such as Hashicorp Vault).
+
+A `SecretClass` looks like this:
+
+[source,yaml]
+----
+apiVersion: secrets.stackable.tech/v1alpha1
+kind: SecretClass
+metadata:
+  name: tls
+spec:
+  backend: # <1>
+    autoTls: # <2>
+      ca:
+        secret:
+          name: secret-provisioner-tls-ca
+          namespace: default
+    # or... <1>
+    k8sSearch: # <3>
+      searchNamespace:
+        pod: {}
+        # or...
+        name: my-namespace
+      secretLabels:
+        type: custom-secret
+----
+<1> Backends are mutually exclusive, only one may be used by each  SecretClass`
+<2> Configures and selects the xref:#backend-autotls[] backend
+<3> Configures and selects the xref:#backend-k8ssearch[] backend
+
+[#backend]
+== Backend
+
+Each `SecretClass` is a associated with a single backend, which dictates the mechanism for issuing that kind of secret.
+
+[#backend-autotls]
+=== `autoTls`
+
+*Format*: xref:#format-tls[]
+
+Issues a TLS certificate signed by the Secret-Operator.
+
+xref:scope.adoc[Scopes] are used to populate the claims (such as `subjectAlternateName`) of the provisioned certificates.
+
+==== Reference
+
+[source,yaml]
+----
+spec:
+  backend:
+    autoTls:
+      ca:
+        secret:
+          name: secret-provisioner-tls-ca
+          namespace: default
+----
+
+`autoTls`:: Declares that the `autoTls` backend is used.
+`autoTls.ca`:: Configures the certificate authority used to issue `Pod` certificates.
+`autoTls.ca.secret`:: Reference (`name` and `namespace`) to a K8s `Secret` object where the CA certificate and key is stored as `ca.crt`
+                      and `ca.key` respectively. Will be created when the first certificate is issued if it does not already exist.
+
+[#backend-k8ssearch]
+=== `k8sSearch`
+
+*Format*: Free-form
+
+A Kubernetes `Secret` object is selected based on the xref:scope.adoc[scopes] specified on the `Volume`. Each field in this `Secret` is mapped
+to one file. It is suggested these `Secret` objects should follow one of the xref:#format[formats defined in this document].
+
+xref:scope.adoc[Scopes] are translated into additional label filters of the form `secrets.stackable.tech/$SCOPE: $SCOPE_VALUE`.
```

Comment:
I read this twice and I still don't understand what it does really.
I think a more "complete" example might help.
One showing a Secret, a Pod and a SecretClass